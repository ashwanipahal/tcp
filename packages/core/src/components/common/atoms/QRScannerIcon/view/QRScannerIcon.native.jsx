import React from 'react';
import CustomIcon from '../../Icon';
import { ICON_NAME, ICON_FONT_CLASS } from '../../Icon/Icon.constants';

const onQRIconFocus = (navigation, closeModal) => {
  if (navigation) {
    navigation.navigate('QRScanner');
  }
  if (closeModal) {
    closeModal();
  }
};

const QRScannerIcon = ({ navigation, closeModal, style, size }) => {
  return (
    <CustomIcon
      onPress={() => onQRIconFocus(navigation, closeModal)}
      iconFontName={ICON_FONT_CLASS.Icomoon}
      name={ICON_NAME.qrcode}
      size={size}
      color="gray.600"
      accessibilityLabel="barcode"
      style={style}
      isButton
    />
  );
};

export default QRScannerIcon;
