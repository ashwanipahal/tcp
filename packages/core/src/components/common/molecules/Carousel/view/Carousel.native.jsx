import React from 'react';
import { TouchableOpacity, View } from 'react-native';
import Carousel from 'react-native-snap-carousel';
import { Image } from '../../../atoms';
import config from '../config';
import theme from '../../../../../../styles/themes/TCP';

const playIcon = require('../../../../../../assets/play.png');
const pauseIcon = require('../../../../../../assets/pause.png');

// @flow
type Props = {
  carouselConfig: Object,
  data: Array<Object>,
  renderItem: Function,
  onSnapToItem: Function,
  width: Number,
  height: Number,
  slideStyle: Object,
};

type State = {
  autoplay: Boolean,
};

/**
 * Default settings for Carousel.
 */
const defaults = { ...config.CAROUSEL_MOBILE_DEFAULTS };

/**
 * Style for play pause icons.
 */
const { colors, zindex } = theme;
const { playIconHeight, playIconWidth } = config.CAROUSEL_MOBILE_CONFIG;
const touchableStyle = {
  backgroundColor: colors.WHITE,
  borderRadius: 15,
  position: 'absolute',
  bottom: 12,
  left: '50%',
  height: playIconHeight,
  width: playIconWidth,
  transform: [
    {
      translateX: -15,
    },
  ],
  zIndex: zindex.zOverlay,
};

/**
 * @function Carousel component that creates carousel using
 * third party library 'react-native-snap-carousel'
 */
class SnapCarousel extends React.PureComponent<Props, State> {
  constructor(props) {
    super(props);
    this.state = {
      autoplay: true,
    };
    this.getPlayButton = this.getPlayButton.bind(this);
    this.play = this.play.bind(this);
    this.pause = this.pause.bind(this);
  }

  /**
   * @function getPlayComponent function gets DOM reference of slider component.
   * @param {[Object]} element [Event object of click].
   * @return {node} function returns slider element.
   */
  getPlayButton() {
    const { autoplay } = this.state;
    return autoplay ? (
      <TouchableOpacity accessibilityRole="button" onPress={this.pause} style={touchableStyle}>
        <Image source={pauseIcon} height={playIconHeight} width={playIconWidth} />
      </TouchableOpacity>
    ) : (
      <TouchableOpacity accessibilityRole="button" onPress={this.play} style={touchableStyle}>
        <Image source={playIcon} height={playIconHeight} width={playIconWidth} />
      </TouchableOpacity>
    );
  }

  /**
   * @function play function enable autoplay for carousel
   * also update component state.
   */
  play() {
    this.carousel.startAutoplay();
    this.togglePlay();
  }

  /**
   * @function pause function stops/pause autoplay for carousel
   * also update component state.
   */
  pause() {
    this.carousel.stopAutoplay();
    this.togglePlay();
  }

  /**
   * @function togglePlay function updates component state
   * after each tap of play pause button.
   */
  togglePlay() {
    const { autoplay } = this.state;
    this.setState({ autoplay: !autoplay });
  }

  render() {
    const {
      carouselConfig: { autoplay: defaultAutoplay },
      data,
      height,
      width,
      onSnapToItem,
      renderItem,
      slideStyle,
    } = this.props;

    return (
      <View>
        <Carousel
          ref={c => {
            this.carousel = c;
          }}
          onSnapToItem={onSnapToItem}
          data={data}
          renderItem={renderItem}
          sliderWidth={width}
          itemWidth={width}
          sliderHeight={height}
          itemHeight={height}
          slideStyle={slideStyle}
          {...defaults}
        />
        {defaultAutoplay && this.getPlayButton()}
      </View>
    );
  }
}

export default SnapCarousel;
