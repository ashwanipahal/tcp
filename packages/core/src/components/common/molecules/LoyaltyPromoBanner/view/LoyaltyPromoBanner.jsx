import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import ClickTracker from '@tcp/web/src/components/common/atoms/ClickTracker';
import { Anchor, RichText, Row, Col } from '../../../atoms';
import { readCookie, setCookie } from '../../../../../utils/cookie.util';
import { loyalityAnalyticsValue } from '../../../../../constants/analytics';
import withStyles from '../../../hoc/withStyles';

import style from '../LoyaltyPromoBanner.style';

/**
 * Check if required UUID cookie available else return default.
 */
const getUUID = uuidCookieString => {
  const UUID = readCookie(uuidCookieString);
  return UUID ? UUID.split(',')[0] : '-1002';
};

/* istanbul ignore next */
const LoyaltyPromoBanner = props => {
  const {
    className,
    cookieID,
    richTextList: [{ richText, link }],
    dataLocator,
  } = props;
  const cookieName = `${cookieID}_${getUUID('WC_USERACTIVITY_')}`;
  const [bannerClosed, setBannerClosed] = useState(true);

  useEffect(() => {
    if (!readCookie(cookieName)) {
      setBannerClosed(false);
    }
  }, [bannerClosed]);

  const closeButtonHandler = () => {
    const currentDate = new Date();
    setCookie({
      key: cookieName,
      value: currentDate.toGMTString(),
      daysAlive: 10,
    });
    setBannerClosed(true);
  };

  return bannerClosed ? null : (
    <div className={`${className} content-wrapper`}>
      <Row>
        <Col
          colSize={{
            small: 6,
            medium: 8,
            large: 12,
          }}
        >
          {link ? (
            <ClickTracker
              as={Anchor}
              to={link.url}
              asPath={link.url}
              target={link.target}
              title={link.title}
              dataLocator={dataLocator || `loyalty-promo-banner`}
              clickData={{
                customEvents: ['event80', 'event81'],
                internalCampaignId: loyalityAnalyticsValue,
              }}
            >
              <RichText richTextHtml={richText.text} />
            </ClickTracker>
          ) : (
            <RichText richTextHtml={richText.text} />
          )}
          <button
            aria-label="close"
            className="loyalty-promo-close-btn"
            onClick={closeButtonHandler}
          >
            <svg
              className="loyalty-promo-close-btn-icon"
              viewBox="0 0 25 25"
              aria-hidden="true"
              role="img"
            >
              <path
                fill="#a0a0a0"
                fillRule="nonzero"
                d="M14.107 12.5l10.56-10.56A1.136 1.136 0 1 0 23.06.333L12.5 10.893 1.94.333A1.136 1.136 0 1 0 .333 1.94l10.56 10.56L.333 23.06a1.136 1.136 0 1 0 1.607 1.607l10.56-10.56 10.56 10.56c.222.222.513.333.804.333a1.136 1.136 0 0 0 .803-1.94L14.107 12.5z"
              />
            </svg>
          </button>
        </Col>
      </Row>
    </div>
  );
};

LoyaltyPromoBanner.propTypes = {
  className: PropTypes.string.isRequired,
  cookieID: PropTypes.string.isRequired,
  richTextList: PropTypes.arrayOf(PropTypes.object),
  dataLocator: PropTypes.string,
};

LoyaltyPromoBanner.defaultProps = {
  richTextList: [
    {
      richText: { text: '' },
      link: {},
    },
  ],
  dataLocator: '',
};

export default withStyles(LoyaltyPromoBanner, style);
export { LoyaltyPromoBanner as LoyaltyPromoBannerVanilla };
