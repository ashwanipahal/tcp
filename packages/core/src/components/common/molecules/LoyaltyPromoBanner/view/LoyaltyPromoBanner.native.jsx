import React from 'react';
import { View } from 'react-native';
import AsyncStorage from '@react-native-community/async-storage';
import PropTypes from 'prop-types';
import { RichText, Anchor } from '../../../atoms';
import { Wrapper, StyledImage, Touchable } from '../LoyaltyPromoBanner.style.native';

const crossImage = require('../../../../../../src/assets/close.png');

const cookieName = '@mprAboveHead_-1002:key';
const date = new Date();
const currentDateValue = date.getTime();
const daysAlive = 10;

class LoyaltyPromoBanner extends React.PureComponent {
  /**
   * To manage the state of icons on the
   * basis of visible & hide .
   */
  constructor(props) {
    super(props);
    this.state = {
      bannerClosed: false,
    };
  }

  componentDidMount() {
    this.isDisplay();
  }

  /**
   * To save the date
   */
  setDate = () => {
    const expiteDateValue = date.setTime(date.getTime() + daysAlive * 24 * 60 * 60 * 1000);
    try {
      AsyncStorage.setItem(cookieName, expiteDateValue.toString());
    } catch (error) {
      // eslint-disable-next-line rule
      console.info('error', error);
    }
  };

  /**
   * To fetch the date
   */
  retrieveDate = async () => {
    try {
      return await AsyncStorage.getItem(cookieName);
    } catch (error) {
      // Error retrieving data
      return null;
    }
  };

  /**
   * To check the visiblity of the banner
   */
  isDisplay = () => {
    this.retrieveDate().then(data => {
      if (data !== null) {
        if (currentDateValue <= parseInt(data, 10)) {
          this.setState({
            bannerClosed: true,
          });
        } else {
          this.setState({
            bannerClosed: false,
          });
        }
      }
    });
  };

  /**
   * This function validate the iconView.
   */
  validateView = () => {
    this.setDate();
    this.setState({ bannerClosed: true });
  };

  /**
   * To render the view
   */
  render() {
    const {
      richTextList: [{ richText, link }],
      navigation,
    } = this.props;

    const { bannerClosed } = this.state;

    return (
      <View>
        {bannerClosed ? (
          <View />
        ) : (
          <Anchor navigation={navigation} url={link.url}>
            <Wrapper>
              <RichText
                source={{
                  html: richText.text,
                }}
              />
              <Touchable
                accessibilityRole="button"
                accessibilityLabels="close"
                onPress={this.validateView}
              >
                <StyledImage source={crossImage} />
              </Touchable>
            </Wrapper>
          </Anchor>
        )}
      </View>
    );
  }
}

LoyaltyPromoBanner.propTypes = {
  richTextList: PropTypes.arrayOf(PropTypes.object),
  navigation: PropTypes.shape({}).isRequired,
};

LoyaltyPromoBanner.defaultProps = {
  richTextList: [
    {
      richText: { text: '' },
      link: {},
    },
  ],
};

export default LoyaltyPromoBanner;
export { LoyaltyPromoBanner as LoyaltyPromoBannerVanilla };
