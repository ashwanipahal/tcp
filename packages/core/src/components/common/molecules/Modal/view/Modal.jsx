/** @module Modal
 *  @summary Wrapper component for react-modal.
 *  Accepts modal content as children and
 *  Other properties of modal to function.
 */

import React from 'react';
import ReactModal from 'react-modal';
import PropTypes from 'prop-types';
import Grid from '../../Grid';
import Row from '../../../atoms/Row';
import Col from '../../../atoms/Col';
import ModalHeader from './ModalHeader';
import styles from '../Modal.style';
import Config from '../Modal.config';
import withStyles from '../../../hoc/withStyles';

class Modal extends React.PureComponent {
  componentDidMount() {
    // Make sure to bind modal to your appElement (http://reactcommunity.org/react-modal/accessibility/)
    const rootElemId = '__next';
    if (process.env.NODE_ENV !== 'test' && document.getElementById(rootElemId)) {
      ReactModal.setAppElement(`#${rootElemId}`);
    }
  }

  render() {
    const { children, ...otherProps } = this.props;
    const {
      colSet,
      onRequestClose,
      title,
      heading,
      fixedWidth,
      className,
      dataLocator,
      dataLocatorHeader,
      closeIconDataLocator,
      headingStyle,
      closeIconLeftAligned,
      ariaLabelledby,
      ariaDescribedby,
      innerContentClassName = '',
      overlayClassName,
      isOpen,
    } = otherProps;
    const column = colSet || Config.MODAL_COL_DEFAULTS;

    return isOpen ? (
      <ReactModal
        {...otherProps}
        overlayClassName={`${className} ${overlayClassName}`}
        aria={{
          labelledby: ariaLabelledby,
          describedby: ariaDescribedby,
        }}
      >
        {!fixedWidth && (
          <Grid>
            <Row>
              <Col
                colSize={column}
                className={`TCPModal__InnerContent ${innerContentClassName}`}
                data-locator={dataLocator}
              >
                <ModalHeader
                  closeFunc={onRequestClose}
                  title={title}
                  heading={heading}
                  closeIconDataLocator={closeIconDataLocator}
                  closeIconLeftAligned={closeIconLeftAligned}
                  dataLocatorHeader={dataLocatorHeader}
                  headingStyle={headingStyle}
                />
                {children}
              </Col>
            </Row>
          </Grid>
        )}
        {fixedWidth && (
          <div
            className={`TCPModal__InnerContent ${innerContentClassName}`}
            data-locator={dataLocator}
          >
            <ModalHeader
              closeFunc={onRequestClose}
              title={title}
              heading={heading}
              closeIconDataLocator={closeIconDataLocator}
              closeIconLeftAligned={closeIconLeftAligned}
              dataLocatorHeader={dataLocatorHeader}
              headingStyle={headingStyle}
            />
            {children}
          </div>
        )}
      </ReactModal>
    ) : null;
  }
}

Modal.propTypes = {
  children: PropTypes.element.isRequired,
};

// TODO removed errorBoundary from modal as its not working properly right now
export default withStyles(Modal, styles);
export { Modal as ModalVanilla };
