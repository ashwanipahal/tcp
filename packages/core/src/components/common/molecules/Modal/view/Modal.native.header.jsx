import React from 'react';
import LineComp from '@tcp/core/src/components/common/atoms/Line';
import ToastContainer from '@tcp/core/src/components/common/atoms/Toast/container/Toast.container.native';
import {
  StyledCrossImage,
  StyledTouchableOpacity,
  ModalHeading,
  LineWrapper,
  RowWrapper,
  ImageWrapper,
  Heading,
} from '../Modal.style.native';
import BodyCopy from '../../../atoms/BodyCopy';

// @flow

type Props = {
  isOpen: boolean,
  stickyCloseIcon: boolean,
  isOverlay?: boolean,
  inheritedStyles?: String,
};

const closeIcon = require('../../../../../assets/close.png');
const arrowIcon = require('../../../../../assets/carrot-large-left.png');

type CloseIconProps = {
  onRequestClose: Function,
  headerStyle: Object,
  iconType: String,
  isOverlay: Boolean,
  rightAlignCrossIcon: Boolean,
  stickyCloseIcon: Boolean,
};

const getCloseIcon = ({
  stickyCloseIcon,
  onRequestClose,
  headerStyle,
  iconType,
  isOverlay,
  rightAlignCrossIcon,
}: CloseIconProps) => {
  return (
    <ImageWrapper stickyCloseIcon={stickyCloseIcon} style={headerStyle}>
      <StyledTouchableOpacity
        onPress={onRequestClose}
        accessibilityRole="button"
        accessibilityLabel="close"
        isOverlay={isOverlay}
      >
        <StyledCrossImage
          rightAlignCrossIcon={rightAlignCrossIcon}
          source={iconType === 'arrow' ? arrowIcon : closeIcon}
        />
      </StyledTouchableOpacity>
    </ImageWrapper>
  );
};

const geLine = (horizontalBar, borderColor) => {
  return (
    <>
      {horizontalBar ? (
        <LineWrapper>
          <LineComp marginTop={5} borderWidth={2} borderColor={borderColor} />
        </LineWrapper>
      ) : null}
    </>
  );
};

const ModalHeaderNative = ({ isOverlay, ...otherProps }: Props) => {
  const {
    heading,
    onRequestClose,
    headingAlign,
    headingFontFamily,
    headerStyle,
    headingFontWeight,
    fontSize,
    iconType,
    fullWidth,
    stickyCloseIcon,
    horizontalBar = true,
    borderColor = 'black',
    rightAlignCrossIcon,
    customHeaderMargin,
  } = otherProps;

  return (
    <Heading>
      <ToastContainer shouldShowSafeArea={false} />
      <RowWrapper
        stickyCloseIcon={stickyCloseIcon}
        isOverlay={isOverlay}
        customHeaderMargin={customHeaderMargin}
      >
        {heading && (
          <ModalHeading stickyCloseIcon={stickyCloseIcon} fullWidth={fullWidth}>
            <BodyCopy
              mobileFontFamily={headingFontFamily || 'primary'}
              fontWeight={headingFontWeight || 'extrabold'}
              textAlign={headingAlign}
              fontSize={fontSize || 'fs16'}
              text={heading}
              numberOfLines={1}
            />
          </ModalHeading>
        )}
        {getCloseIcon({
          onRequestClose,
          headerStyle,
          iconType,
          isOverlay,
          stickyCloseIcon,
          rightAlignCrossIcon,
        })}
      </RowWrapper>
      {geLine(horizontalBar, borderColor)}
    </Heading>
  );
};

ModalHeaderNative.defaultProps = {
  isOverlay: false,
  inheritedStyles: '',
};

export default ModalHeaderNative;
