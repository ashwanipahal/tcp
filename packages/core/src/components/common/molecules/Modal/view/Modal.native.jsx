import React from 'react';
import {
  Modal,
  StatusBar,
  SafeAreaView,
  ScrollView,
  Platform,
  KeyboardAvoidingView,
} from 'react-native';
import LineComp from '@tcp/core/src/components/common/atoms/Line';
import ToastContainer from '@tcp/core/src/components/common/atoms/Toast/container/Toast.container.native';
import {
  StyledCrossImage,
  StyledTouchableOpacity,
  ModalHeading,
  LineWrapper,
  RowWrapper,
  ImageWrapper,
  ModalCustomWrapper,
} from '../Modal.style.native';
import BodyCopy from '../../../atoms/BodyCopy';

// How To use this react native modal
// import this component in your file.
// It takes four properties: isOpen, onRequestClose, heading and children

// @flow

type Props = {
  isOpen: boolean,
  stickyCloseIcon: boolean,
  children: node,
  isOverlay?: boolean,
  inheritedStyles?: String,
};

const closeIcon = require('../../../../../assets/close.png');
const arrowIcon = require('../../../../../assets/carrot-large-left.png');

type CloseIconProps = {
  onRequestClose: Function,
  headerStyle: Object,
  iconType: String,
  isOverlay: Boolean,
  stickyCloseIcon: Boolean,
};

const getCloseIcon = ({
  stickyCloseIcon,
  onRequestClose,
  headerStyle,
  iconType,
  isOverlay,
}: CloseIconProps) => {
  return (
    <ImageWrapper stickyCloseIcon={stickyCloseIcon} style={headerStyle}>
      <StyledTouchableOpacity
        onPress={onRequestClose}
        accessibilityRole="button"
        accessibilityLabel="close"
        isOverlay={isOverlay}
      >
        <StyledCrossImage source={iconType === 'arrow' ? arrowIcon : closeIcon} />
      </StyledTouchableOpacity>
    </ImageWrapper>
  );
};

const geLine = (horizontalBar, borderColor) => {
  return (
    <>
      {horizontalBar ? (
        <LineWrapper>
          <LineComp marginTop={5} borderWidth={2} borderColor={borderColor} />
        </LineWrapper>
      ) : null}
    </>
  );
};

const ModalNative = ({ isOpen, children, isOverlay, inheritedStyles, ...otherProps }: Props) => {
  const {
    heading,
    onRequestClose,
    animationType,
    headingAlign,
    headingFontFamily,
    headerStyle,
    headingFontWeight,
    fontSize,
    iconType,
    fullWidth,
    customTransparent,
    stickyCloseIcon,
    transparentModal,
    horizontalBar = true,
    borderColor = 'black',
  } = otherProps;
  let behavior = null;
  let keyboardVerticalOffset = 0;
  if (Platform.OS === 'ios') {
    behavior = 'padding';
    keyboardVerticalOffset = 64;
  }

  return (
    <SafeAreaView>
      <Modal
        transparent={customTransparent || false}
        visible={isOpen}
        animationType={animationType}
        onRequestClose={onRequestClose}
      >
        {!customTransparent && (
          <ModalCustomWrapper transparentModal={transparentModal} inheritedStyles={inheritedStyles}>
            <ToastContainer />
            <StatusBar hidden />
            <RowWrapper stickyCloseIcon={stickyCloseIcon} isOverlay={isOverlay}>
              {heading && (
                <ModalHeading stickyCloseIcon={stickyCloseIcon} fullWidth={fullWidth}>
                  <BodyCopy
                    mobileFontFamily={headingFontFamily || 'primary'}
                    fontWeight={headingFontWeight || 'extrabold'}
                    textAlign={headingAlign}
                    fontSize={fontSize || 'fs16'}
                    text={heading}
                  />
                </ModalHeading>
              )}
              {getCloseIcon({
                onRequestClose,
                headerStyle,
                iconType,
                isOverlay,
                stickyCloseIcon,
              })}
            </RowWrapper>
            {geLine(horizontalBar, borderColor)}
            <KeyboardAvoidingView
              behavior={behavior}
              keyboardVerticalOffset={keyboardVerticalOffset}
              enabled
            >
              <ScrollView showsVerticalScrollIndicator={false} keyboardShouldPersistTaps="handled">
                {children}
              </ScrollView>
            </KeyboardAvoidingView>
          </ModalCustomWrapper>
        )}
        {customTransparent && children}
      </Modal>
    </SafeAreaView>
  );
};

ModalNative.defaultProps = {
  isOverlay: false,
  inheritedStyles: '',
};

export default ModalNative;
