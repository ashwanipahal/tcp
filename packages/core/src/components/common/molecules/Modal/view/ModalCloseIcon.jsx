import React from 'react';
import { css } from 'styled-components';
import PropTypes from 'prop-types';
import withStyles from '../../../hoc/withStyles';

const CloseButtonStyle = css`
  background: transparent url('/static/images/modal-close.svg') no-repeat 0 0;
  border: none;
  cursor: pointer;
  position: absolute;
  padding: 0px 7px 13px 6px;
  background-size: 100% 100%;
  width: 15px;
  height: 15px;
  &.alignRight {
    right: 15px;
  }
  &.alignLeft {
    left: 15px;
  }
  &.alignTop {
    top: ${props => props.theme.spacing.LAYOUT_SPACING.XXS};
  }
`;

const ModalCloseIcon = ({ className, closeFunc, closeIconDataLocator, closeIconLeftAligned }) => (
  <button
    aria-label="close"
    tabIndex="0"
    className={`${className} ${'alignTop'} ${closeIconLeftAligned ? 'alignLeft' : 'alignRight'}`}
    onClick={e => closeFunc(e)}
    data-locator={closeIconDataLocator}
  />
);

ModalCloseIcon.propTypes = {
  className: PropTypes.string.isRequired,
  closeFunc: PropTypes.func.isRequired,
  closeIconDataLocator: PropTypes.string,
  closeIconLeftAligned: PropTypes.bool,
};

ModalCloseIcon.defaultProps = {
  closeIconDataLocator: 'close',
  closeIconLeftAligned: false,
};

export default withStyles(ModalCloseIcon, CloseButtonStyle);
export { ModalCloseIcon as ModalCloseIconVanilla };
