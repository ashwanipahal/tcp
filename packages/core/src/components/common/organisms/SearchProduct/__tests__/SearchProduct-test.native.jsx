import React from 'react';
import { shallow } from 'enzyme';

import { SearchProductVanilla } from '../views/SearchProduct.view.native';

describe('SearchProduct should render correctly', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SearchProductVanilla />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should close modal', () => {
    const closeSearchModal = jest.fn();
    wrapper.setProps({ closeSearchModal });
    wrapper.instance().closeModal();
    expect(wrapper.instance().state.modalVisible).toBeFalsy();
    expect(closeSearchModal).toHaveBeenCalled();
  });

  it('should start search on change text', () => {
    const startSearch = jest.fn();
    wrapper.setProps({ startSearch });
    wrapper.instance().onChangeText('p');
    expect(startSearch).toHaveBeenCalledTimes(0);
    wrapper.instance().onChangeText('pant');
    expect(startSearch).toHaveBeenCalledTimes(1);
    expect(wrapper.instance().state.searchText).toBe('pant');
  });

  it('should go to srp on search', () => {
    const goToSearchResultsPage = jest.fn();
    wrapper.setProps({ goToSearchResultsPage });
    wrapper.instance().searchProducts('pants');
    expect(wrapper.instance().state.modalVisible).toBeFalsy();
    expect(goToSearchResultsPage).toHaveBeenCalled();
  });
});
