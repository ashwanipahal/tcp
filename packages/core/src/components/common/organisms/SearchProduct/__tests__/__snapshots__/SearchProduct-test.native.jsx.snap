// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchProduct should render correctly should match snapshot 1`] = `
<Component
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={false}
  visible={true}
>
  <Styled(SafeAreaView)>
    <Styled(View)>
      <Styled(CustomIcon)
        accessibilityLabel=""
        color="blue.500"
        dataLocator=""
        iconFontName="Icomoon"
        isButton={false}
        isDisabled={false}
        margin={null}
        name="mag-glass"
        onPress={[Function]}
        size="fs25"
      />
      <Styled(TextInput)
        autoFocus={true}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        returnKeyType="search"
        value=""
      />
      <Styled(CustomIcon)
        accessibilityLabel=""
        color="gray.600"
        dataLocator=""
        iconFontName="Icomoon"
        isButton={false}
        isDisabled={false}
        margin={null}
        name="scan"
        onPress={[Function]}
        size="fs25"
      />
      <Styled(CustomButton)
        active={false}
        buttonVariation="fixed-width"
        color=""
        customStyle={Object {}}
        customTextStyle={
          Object {
            "fontWeight": "normal",
            "textTransform": "none",
          }
        }
        disableButton={false}
        fill=""
        fontFamily="secondary"
        fontSize="fs13"
        fontWeight="regular"
        fullWidth=""
        iconColor="gray.800"
        iconName="chevron-down"
        iconSize="fs12"
        locator=""
        navigation={Object {}}
        onPress={[Function]}
        selected={false}
        selectedIcon="chevron-up"
        showIcon={false}
        style={
          Object {
            "borderWidth": 0,
          }
        }
        text="Cancel"
        theme={Object {}}
        url=""
        width="90px"
      />
    </Styled(View)>
    <SectionList
      data={Array []}
      disableVirtualization={false}
      extraData=""
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      keyboardShouldPersistTaps="handled"
      maxToRenderPerBatch={10}
      onEndReachedThreshold={2}
      renderItem={[Function]}
      renderSectionHeader={[Function]}
      scrollEventThrottle={50}
      sections={Array []}
      stickySectionHeadersEnabled={false}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </Styled(SafeAreaView)>
</Component>
`;
